vivo-project/VIVO
# Enable war deployment model

Partial resolution to: https://jira.lyrasis.org/browse/VIVO-1443

# What does this pull request do?
This update changes VIVO to no longer post the (exploded) war to Tomcat during build, nor create the VIVO_HOME directories.
Instead, the build now produces a war file, `vivo.war` (found in "installer/webapp/target/") that must be "manually" installed by copying it into the Tomcat webapps directory.
When Tomcat/VIVO is started, the VIVO_HOME directory is populated from the standard resources that are now built into the war file.
Other notes:
- Hardcode vivo.all.log file name
- Must now set system property on application deploy: `-Dvivo-dir=/opt/vivo/home/`
- Optionally provide build property to set app-name: `-Dapp-name=vivo`

Once the process for building, deploying and customizing has been tested with this pull-request, I would like to follow-on by simplifying the Maven structure by removing the various "installer" Maven modules.

# How should this be tested?
To test, 
1. Build VIVO and notice that files in VIVO_HOME have not been changed
   - Update your Tomcat system properties (/etc/default/tomcat9 ?) to include the `vivo.dir` system property that references an existing VIVO_HOME directory, such as:
      ```
      JAVA_OPTS="${JAVA_OPTS} -Dvivo-dir=/opt/vivo/home/"
      ```
   - Install VIVO by dropping the generated war file into the Tomcat webapps directory
2. If your VIVO_HOME directory exists but is empty, it will be populated with the default files, including `default.runtime.properties`, `default.applicationSetup.n3` and `default.developer.properties`.
   - If your VIVO_HOME directory exists and is populated, standard files in bin and rdf directories will be updated.
   - If it is desired to keep changes in VIVO_HOME directory you will need to create a checksum digest that is used to determine whether or not to overwrite. Otherwise, the digest is created first deploy if not yet exists.
   
Command to create checksum digest.
```
find /vivo/home -type f | cut -c3- | grep -E '^bin/|^config/|^rdf/' | xargs md5sum > /vivo/home/digest.md5`
```

For testing with a third-tier build, supporting overlay of webapp and home directory.
https://github.com/awoods/vivo-custom-webapp/pull/2

# Interested parties
@VIVO-project/vivo-committers
